--- js/public/experimental/CompileScript.h.orig 2024-03-11 15:30:52.076176490 +0000
+++ js/public/experimental/CompileScript.h      2024-03-11 15:32:01.913707237 +0000
@@ -50,7 +50,7 @@
 //
 // Off-thread consumers can allocate this instance off main thread, and pass it
 // back to the main thread, in order to reduce the main thread allocation.
-struct CompilationStorage {
+struct JS_PUBLIC_API CompilationStorage {
  private:
   // Owned CompilationInput.
   //

--- js/public/experimental/JSStencil.h.orig     2024-03-11 15:30:52.078071703 +0000
+++ js/public/experimental/JSStencil.h  2024-03-11 15:32:01.915428406 +0000
@@ -48,7 +48,7 @@

 namespace JS {

-struct CompilationStorage;
+struct JS_PUBLIC_API CompilationStorage;

 using Stencil = js::frontend::CompilationStencil;
 using FrontendContext = js::FrontendContext;
