--- build/moz.configure/toolchain.configure.orig	2024-04-09 14:09:50
+++ build/moz.configure/toolchain.configure	2024-04-09 14:11:04
@@ -1752,7 +1752,7 @@
             die("Unsupported linker " + linker)

         # Check the kind of linker
-        version_check = ["-Wl,--version"]
+        version_check = ["-Wl,-ld_classic,--version"]
         cmd_base = c_compiler.wrapper + [c_compiler.compiler] + c_compiler.flags

         def try_linker(linker):
